src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/RecommenderPackageFeed.cs

```diff
@@ -86,18 +90,27 @@ private class RecommendSearchToken : ContinuationToken
             public SearchFilter SearchFilter { get; set; }
         }
 
-        public Task<SearchResult<IPackageSearchMetadata>> SearchAsync(string searchText, SearchFilter searchFilter, CancellationToken cancellationToken)
+        public async Task<SearchResult<IPackageSearchMetadata>> SearchAsync(string searchText, SearchFilter searchFilter, CancellationToken cancellationToken)
         {
             var searchToken = new RecommendSearchToken
             {
                 SearchString = searchText,
                 SearchFilter = searchFilter,
             };
 
-            return RecommendPackagesAsync(searchToken, cancellationToken);
+            SearchResult<IPackageSearchMetadata> recommenderResults = await RecommendPackagesAsync(searchToken, cancellationToken);
+            SearchResult<IPackageSearchMetadata> decoratedResults = await _decoratedPackageFeed.SearchAsync(searchText, searchFilter, cancellationToken);
+
+            // Add the recommended results to the top of the decorated feed's results, deduplicating any packages in the decorated feed.
+            IReadOnlyList<IPackageSearchMetadata> combinedResults = recommenderResults.Items.Union(decoratedResults.Items, new IdentityIdEqualityComparer()).ToList();
```

Comment:
Suggestion: Keep a static property with the equality comparer to avoid allocations on each search.